### PR TITLE
added FindGSL and updated README.md accordingly

### PR DESCRIPTION
**Pull request by protobits**
The .cmake file was taken from: http://sourceforge.net/p/openmodeller/svn/HEAD/tree/trunk/openmodeller/cmake/FindGSL.cmake
The original GSL code does not provide a .cmake file. It would be useful to include this here.

### CODE REVIEW

File: cmake/Modules/FindGSL.cmake [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+# Once run this will define:
+#
+# GSL_FOUND       = system has GSL lib
+#
+# GSL_LIBRARIES   = full path to the libraries
+#    on Unix/Linux with additional linker flags from "gsl-config --libs"
+#
+# CMAKE_GSL_CXX_FLAGS  = Unix compiler flags for GSL, essentially "`gsl-config --cxxflags`"
+#
+# GSL_INCLUDE_DIR      = where to find headers
```
Comment (wjwwood) [inline cmake/Modules/FindGSL.cmake]: This should be `GSL_INCLUDE_DIRS`, even if the path is singular. See:

https://github.com/ros/cmake_modules/blob/0.3-devel/CONTRIBUTING.md#cmake-module-style

**Comment by protobits** [comment]
I've changed all occurrences, is that OK? I'm not sure if any of these should be singular.

**Comment by wjwwood** [comment]
That looks OK, but another way to do it, which doesn't interfere with the original code as much, is to just change the documentation and then do something like `set(GSL_INCLUDE_DIRS ${GSL_INCLUDE_DIR})` at the end of the module. Either way works, just thought I'd let you know that was an option too. Just let me know if you want to change it again or merge as is.

Thanks.

**Comment by protobits** [comment]
I dont mind the change. You can merge as is.

**Pull request merged**